### PR TITLE
Remove Automatic Snapshot Deployment to GCR for older versions

### PR DESCRIPTION
**Pull request by matrei**
Only the latest release branch should automatically deploy snapshots to https://start.grails.org.

### CODE REVIEW

File: .github/workflows/snapshot.yml [hunk 1]
```diff
@@ -1,11 +1,9 @@
 name: Snapshot
 on:
-  push:
```
Comment (jdaugherty) [inline .github/workflows/snapshot.yml:3]: Shouldn't the CLI still be published as a snapshot?  It's only the google deploy we want to avoid ... right?

**Review by jdaugherty — CHANGES_REQUESTED** [review]

File: .github/workflows/snapshot.yml [hunk 2]
```diff
@@ -9,287 +9,330 @@ on:
...
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
-        with:
-          arguments: publishToSonatype
+        run: ./gradlew publishToSonatype
   deploy:
-    name: Deploy To Google Cloud Run
+    # Snapshots should only be deployed from the latest release branch
+    # as it updates the snapshot version on https://start.grails.org.
+    if: false # github.event_name != 'pull_request'
```
Comment (jdaugherty) [inline .github/workflows/snapshot.yml:54]: Is there a reason we're leaving the deploy step here if we're going to hard code it to false?  From what I can tell you moved it to the release.yml step, how do we ensure the latest 7.0.0 build gets built if it's not released yet?

Comment (matrei) [reply .github/workflows/snapshot.yml:54]: > Is there a reason we're leaving the deploy step here if we're going to hard code it to false?

Well, I could remove it instead but I thought it was self-documenting having it there. And if you, for some reason, did need to deploy, you and just flip the switch.

Would you rather have me remove it?

> From what I can tell you moved it to the release.yml step

 I don't think I moved it to `release.yml`. It was already there.

> how do we ensure the latest 7.0.0 build gets built if it's not released yet?

I'm not sure I understand the question. This PR is for the 6.0.x branch. I am planning to merge it up to 6.1.x and 6.2.x making changes as necessary. The 7.0.x branch will not get any changes to GCR publishing at the moment.

Comment (jdaugherty) [reply .github/workflows/snapshot.yml:54]: I understand now, lets leave it?

**Review by jdaugherty — APPROVED** [review]

**Pull request merged**